src/Redistribution.sol

```diff
@@ -594,10 +669,6 @@ contract Redistribution is AccessControl, Pausable {
             return currentSeed();
         }
 
-        if (currentPhaseReveal() && cr == currentRevealRound) {
-            require(false, "can't return value after first reveal");
```

Comment:
you just removed this but it was defined here because it has the same ambiguity as invoking `isParticipatingInUpcomingRound` in the reveal phase, hasn't it?

Reply:
I thought about this way more than I should now :) Going back and forth and putting it back and then again removing it.

So currentRoundAnchor is used here and with this it will never fire that condition.
![image](https://github.com/ethersphere/storage-incentives/assets/8969767/7d4f5630-5313-4a11-a023-eac75690dbbc)

For isParticipatingInUpcomingRound  I am not sure why do we want to have it there.  What this does is that we can check anchor in reveal round but before first reveal. And with that isParticipatingInUpcomingRound can be checked in reveal round before first reveal. I am wondering why we want to enable checking anchor and isParticipatingInUpcomingRound at the point before first reveal in reveal phase?

Seems obsolete to me. But maybe there is some case for it in bee node somehow?
(putting it back again now just in case I dont break bee node, but still wondering why do we need this? )

Reply:
Ok I am trying to get why this is needed in that situation. Could be I dont know some info, seems strange to me to check for participation in the upcoming round in the revealed round running.  I added it, but tried to make it more into context as it is used for isParticipatingInUpcomingRound for some still unknown reason to me, so I added it there as check.

Reply:
Hmm starting to get what they wanted with this. To not show next anchor in reveal phase, as it is set for next round after first reveal with new seed. I think it kind of convoluted this way. Will see If i can make it more clear and still keep the same idea.

Reply:
I made this and just realized, this also doesnt make any sense as it will never get to that point, there is FIRST IF that will revert and will not get to that. So there is really no point in that code.

![image](https://github.com/ethersphere/storage-incentives/assets/8969767/c72fa67d-2adc-4ee6-8163-088bc9dbd4e4)